### PR TITLE
Improve problem aggregation

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes 
 - #28215 

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test this

File: platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/protocol/InternalProblemAggregationV2.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.gradle.api.NonNullApi;
+import org.gradle.tooling.internal.protocol.problem.InternalDocumentationLink;
+import org.gradle.tooling.internal.protocol.problem.InternalSeverity;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@NonNullApi
+public interface InternalProblemAggregationV2 extends InternalProblemAggregation {
```
Comment (donat) [inline platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/protocol/InternalProblemAggregationV2.java:27]: ❌ To follow the pattern in the `protocol` package, this interface should be called `InternalProblemAggregationVersion2`.

File: platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemContextDetails.java [hunk 2]
```diff
@@ -27,66 +27,33 @@
...
-    ProblemCategory getCategory();
-
-    /**
-     * Returns the problem label.
-     *
-     * @return the problem label
-     * @since 8.6
-     */
-    Label getLabel();
+public interface ProblemContextDetails {
```
Comment (donat) [inline platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemContextDetails.java:33]: Can we call it `ProblemContext`?

File: platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemAggregation.java [hunk 3]
```diff
@@ -47,12 +48,29 @@ public interface ProblemAggregation {
...
+
     /**
      * The list of aggregated problems.
      *
      * @return The list of aggregated problems.
-     * @since 8.6
+     * @since 8.8
      */
-    List<ProblemDescriptor> getProblemDescriptors();
+    List<ProblemContextDetails> getProblemContextDetails();
```
Comment (donat) [inline platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemAggregation.java:74]: getContexts()
getProblemContexts()?

File: platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemAggregationEvent.java [hunk 4]
```diff
@@ -32,10 +30,10 @@
 public interface ProblemAggregationEvent extends ProblemEvent {
 
     /**
-     * Returns the list of aggregated problems.
-     * @return The list of aggregated problems.
+     * Returns the problem category that is common for all aggregated problems.
```
Comment (donat) [inline platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemAggregationEvent.java:33]: ❌ This doesn't seem accurate.

**Review by donat — CHANGES_REQUESTED** [review]
Looks good, I mainly have review items about naming and documentation.

**Review by donat — APPROVED** [review]

**Pull request merged**